### PR TITLE
Add `IsOptimistic` chain info getter

### PR DESCRIPTION
**Pull request by terencechain**
Add `IsOptimistic` chain info getter

### CODE REVIEW

File: beacon-chain/blockchain/testing/mock.go [hunk 1]
```diff
@@ -438,3 +438,8 @@ func (s *ChainService) HeadSyncSelectionProofDomain(_ context.Context, _ types.S
 func (s *ChainService) HeadSyncContributionProofDomain(_ context.Context, _ types.Slot) ([]byte, error) {
 	return s.SyncContributionProofDomain, nil
 }
+
+// IsOptimistic mocks the same method in the chain service.
+func (s *ChainService) IsOptimistic(ctx context.Context) (bool, error) {
```
Comment (potuz) [inline beacon-chain/blockchain/testing/mock.go:443]:
```suggestion
func (s *ChainService) IsOptimistic(_ context.Context) (bool, error) {
```
from DeepSource

**Review by potuz — APPROVED** [review]
LGTM

File: beacon-chain/blockchain/chain_info.go [hunk 2]
```diff
@@ -328,6 +329,13 @@ func (s *Service) HeadValidatorIndexToPublicKey(_ context.Context, index types.V
 	return v.PublicKey(), nil
 }
 
+// IsOptimistic returns true if the current head is optimistic.
+func (s *Service) IsOptimistic(ctx context.Context) (bool, error) {
+	s.headLock.RLock()
+	defer s.headLock.RUnlock()
+	return s.cfg.ForkChoiceStore.Optimistic(ctx, s.head.root, s.head.slot)
```
Comment (jmozah) [inline beacon-chain/blockchain/chain_info.go:336]: Should we check only this head OR we test every tip of synced_tips to be optimistic and then say optimistic?

Comment (terencechain) [reply beacon-chain/blockchain/chain_info.go:336]: Only the head

**Pull request merged**